usr/lib/soft_stdll/soft_specific.c

```diff
@@ -320,6 +369,53 @@ CK_RV token_specific_des_cbc(STDLL_TokData_t *tokdata,
     }
 
     return rc;
+#else
+    const EVP_CIPHER *cipher = EVP_des_cbc();
+    EVP_CIPHER_CTX *ctx = NULL;
+    CK_ATTRIBUTE *attr = NULL;
+    unsigned char dkey[8];
+    CK_ULONG rc;
+    int outlen;
+
+    UNUSED(tokdata);
+
+    // get the key value
+    if (template_attribute_find(key->template, CKA_VALUE, &attr) == FALSE) {
+        TRACE_ERROR("template_attribute_find(CKA_VALUE) failed.\n");
+        return CKR_FUNCTION_FAILED;
+    }
+
+    memcpy(dkey, attr->pValue, sizeof(dkey));
+
+    if (in_data_len % 8 || in_data_len > INT_MAX) {
```

Comment:
move this block befoe the memcpy of the key

Reply:
done